jackson-databind/src/test/groovy/io/micronaut/jackson/modules/BeanIntrospectionModuleSpec.groovy

```diff
@@ -354,6 +354,7 @@ class BeanIntrospectionModuleSpec extends Specification {
     void "test that introspected serialization works with @JsonAnyGetter"() {
```

Comment:
I'm confused.  This test used to pass...  Wasn't it testing that `@JsonAnyGetter` worked as expected?

Reply:
There are two branches in the BIM. One branch "enhances" the properties discovered by jackson using reflection by replacing the property getters/setters with non-reflective BeanIntrospection access. The second branch runs when jackson discovered no properties using reflection, which is typically the case with native image (but can also happen when there are simply no normal properties). The first branch worked fine and was covered by this test already, the second branch is fixed by this PR.